File: src/python/algorithms/array/equillibriumIndex/equilibrium_index_test.py
import unittest
from equilibrium_index import equilibrium_index

class TestEquilibriumIndex(unittest.TestCase):
    def test_equilibrium_index_found(self):
        self.assertEqual(equilibrium_index([1, 3, 5, 2, 2]), 2)

    def test_no_equilibrium_index(self):
        self.assertEqual(equilibrium_index([1, 2, 3]), -1)

    def test_single_element(self):
        self.assertEqual(equilibrium_index([10]), 0)

    def test_empty_array(self):
        self.assertEqual(equilibrium_index([]), -1)

if __name__ == '__main__':
    unittest.main()

File: src/java1/algorithms/array/equillibriumIndex/EquilibriumIndexTest.java
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;
public class EquilibriumIndexTest {
    @Test
    void testEquilibriumIndexFound() {
        int[] nums = {1, 3, 5, 2, 2};
        assertEquals(2, EquilibriumIndex.equilibriumIndex(nums));
    }

    @Test
    void testNoEquilibriumIndex() {
        int[] nums = {1, 2, 3};
        assertEquals(-1, EquilibriumIndex.equilibriumIndex(nums));
    }

    @Test
    void testSingleElement() {
        int[] nums = {10};
        assertEquals(0, EquilibriumIndex.equilibriumIndex(nums));
    }

    @Test
    void testEmptyArray() {
        int[] nums = {};
        assertEquals(-1, EquilibriumIndex.equilibriumIndex(nums));
    }
}